crates/dojo-world/src/manifest.rs

```diff
@@ -121,118 +104,197 @@ pub struct Contract {
 #[serde_as]
 #[derive(Clone, Default, Debug, Serialize, Deserialize, PartialEq)]
 pub struct Class {
-    pub name: SmolStr,
     #[serde_as(as = "UfeHex")]
     pub class_hash: FieldElement,
-    pub abi: Option<abi::Contract>,
+    pub abi: Option<String>,
 }
 
-#[derive(Clone, Default, Debug, Serialize, Deserialize, PartialEq)]
+#[serde_as]
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug)]
 pub struct Manifest {
-    pub world: Contract,
-    pub executor: Contract,
-    pub base: Class,
-    pub contracts: Vec<Contract>,
-    pub models: Vec<Model>,
+    pub kind: ManifestKind,
+    pub name: SmolStr,
 }
 
 impl Manifest {
-    /// Load the manifest from a file at the given path.
-    pub fn load_from_path(path: impl AsRef<Path>) -> Result<Self, std::io::Error> {
-        let file = fs::File::open(path)?;
-        Ok(Self::try_from(file)?)
+    pub fn class_hash(&self) -> FieldElement {
+        match &self.kind {
+            ManifestKind::Class(class) => class.class_hash,
+            ManifestKind::Contract(contract) => contract.class_hash,
+            ManifestKind::Model(model) => model.class_hash,
+        }
+    }
+
+    pub fn abi(&self) -> Option<&String> {
+        match &self.kind {
+            ManifestKind::Class(class) => class.abi.as_ref(),
+            ManifestKind::Contract(contract) => contract.abi.as_ref(),
+            ManifestKind::Model(model) => model.abi.as_ref(),
+        }
     }
 
-    /// Writes the manifest into a file at the given path. Will return error if the file doesn't
-    /// exist.
-    pub fn write_to_path(self, path: impl AsRef<Path>) -> Result<(), std::io::Error> {
-        let fd = fs::File::options().write(true).open(path)?;
-        Ok(serde_json::to_writer_pretty(fd, &self)?)
+    pub fn set_class_hash(&mut self, class_hash: FieldElement) {
+        match &mut self.kind {
+            ManifestKind::Class(class) => class.class_hash = class_hash,
+            ManifestKind::Contract(contract) => contract.class_hash = class_hash,
+            ManifestKind::Model(model) => model.class_hash = class_hash,
+        }
     }
 
-    /// Construct a manifest of a remote World.
-    ///
-    /// # Arguments
-    /// * `provider` - A Starknet RPC provider.
-    /// * `world_address` - The address of the remote World contract.
-    pub async fn load_from_remote<P>(
+    pub fn set_abi(&mut self, abi: Option<String>) {
+        match &mut self.kind {
+            ManifestKind::Class(class) => class.abi = abi,
+            ManifestKind::Contract(contract) => contract.abi = abi,
+            ManifestKind::Model(model) => model.abi = abi,
+        }
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum ManifestKind {
+    Class(Class),
+    Contract(Contract),
+    Model(Model),
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct World {
```

Comment:
Or you didn't want to use the word `Manifest` as this struct is an "abstract" manifest and not a file on it's own?

Reply:
Yes i agree, `WorldManifest` wouldn't work because there is already a field called `world` inside this manifest.

> Or you didn't want to use the word Manifest as this struct is an "abstract" manifest and not a file on it's own?

also this yes

maybe we call it `AbstractManifest`? i am open for suggestions on this.